examples/lib/dynamic_outputs/pipeline.ex

```diff
@@ -0,0 +1,194 @@
+    {update_scene_action(inputs, outputs), %{state | inputs: inputs, outputs: outputs}}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:new_output_stream, output_id, _vc_ctx},
+        :video_compositor,
+        _membrane_ctx,
+        state
+      ) do
+    Process.send_after(self(), {:remove_output, output_id}, 10_000)
+    {[spec: output_spec(output_id)], state}
+  end
+
+  @impl true
+  def handle_child_notification(
+        {:vc_request_response, req, %Req.Response{status: response_code, body: response_body},
+         _vc_ctx},
+        :video_compositor,
+        _membrane_ctx,
+        state
+      ) do
+    if response_code != 200 do
+      raise """
+      Request failed.
+      Request: `#{inspect(req)}.
+      Response code: #{response_code}.
+      Response body: #{inspect(response_body)}.
+      """
+    end
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_child_notification(notification, child, _ctx, state) do
+    Membrane.Logger.info(
+      "Received notification: #{inspect(notification)} from child: #{inspect(child)}."
+    )
+
+    {[], state}
+  end
+
+  @impl true
+  def handle_info({:register_output, output_id}, _ctx, state) do
+    output_opt = %OutputOptions{
+      width: @output_width,
+      height: @output_height,
+      id: output_id
+    }
+
+    {[notify_child: {:video_compositor, {:register_output, output_opt}}], state}
+  end
+
+  @impl true
+  def handle_info(
+        {:remove_output, output_id},
+        _ctx,
+        state = %{inputs: inputs, outputs: outputs}
+      ) do
+    outputs = outputs |> Enum.reject(fn id -> id == output_id end)
+
+    children = [
+      {:output_parser, output_id},
+      {:output_decoder, output_id},
+      {:sdl_player, output_id}
+    ]
+
+    # Change the scene before removing the output.
+    # VideoCompositor forbids removing output used in the current scene.
+    {update_scene_action(inputs, outputs) ++ [remove_children: children],
+     %{state | outputs: outputs}}
+  end
+
+  @spec update_scene_action(list(VideoCompositor.input_id()), list(VideoCompositor.output_id())) ::
+          [Membrane.Pipeline.Action.notify_child()]
+  defp update_scene_action(input_ids, output_ids) do
+    update_scene_request =
+      if length(output_ids) > 0 and length(input_ids) > 0 do
+        outputs =
+          output_ids
+          |> Enum.map(fn output_id -> %{output_id: output_id, input_pad: @layout_id} end)
+
+        %{
+          type: :update_scene,
+          nodes: [tiled_layout(input_ids)],
+          outputs: outputs
+        }
+      else
+        %{
+          type: :update_scene,
+          nodes: [],
+          outputs: []
+        }
+      end
+
+    [notify_child: {:video_compositor, {:vc_request, update_scene_request}}]
+  end
+
+  @spec tiled_layout(list(String.t())) :: map()
+  defp tiled_layout(input_pads) do
+    %{
+      type: "built-in",
+      transformation: :tiled_layout,
+      node_id: "layout",
```

Comment:
I wouldn't say I like mixing strings and atoms like this. When we release VC in the new version, where `"builtin"` doesn't use `"-"` in the middle, I'll replace all `type:` fields with atoms. Leaving `node_id` as a string is a good idea since they shouldn't be dynamically created in runtime (they aren't garbage collected, the beam has a limit on the number of created atoms, etc.).